ethpm/package.py

```diff
@@ -84,15 +84,24 @@ def version(self):
     # Deployments
     #
 
-    def get_deployments(self, w3):
+    def get_deployments(self, w3=None):
         """
         API to retrieve instance of deployed contract dependency.
         """
+        if w3 is None:
+            w3 = self.w3
+
         validate_w3_instance(w3)
         validate_deployments_are_present(self.package_data)
 
         all_blockchain_uris = self.package_data["deployments"].keys()
         matching_uri = validate_single_matching_uri(all_blockchain_uris, w3)
 
         deployment_data = self.package_data["deployments"][matching_uri]
-        return Deployments(deployment_data, w3)
+
+        all_contract_factories = {}
+        all_contract_names = list(self.package_data["contract_types"].keys())
+        for name in all_contract_names:
+            all_contract_factories[name] = self.get_contract_type(name, w3)
```

Comment:
You can use a dictionary comprehension

```python
all_contract_factories = {
    name: self.get_contract_type(name, w3) 
    for name in 
    self.package_data["contract_types"].keys()
}
```

Note that `package_data` isn't required to have a `contract_types` key.  Also, it may be prudent to not put *all* contract types in, but rather to only put in the types necessary for the deployments.

```python
deployments = self.package_data["deployments"][matching_uri]
all_contract_factories = {
    deployment_data['contract_type']: self.get_contract_type(deployment_data['contract_type'], w3) 
    for deployment_data
    in deployments.values()
}
```

This way you are passing in the absolute minimal information necessary for the `Deployments` object to do it's magic.